### PR TITLE
[quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35688 [quant] Move quantization tests into test/quantization
* #35571 [quant][graphmode] Quantization support for at::sort
* #35558 [quant][graphmode][refactor] getGeneralTensorInputs(Node*) -> getPassThroughInputs(Value*)
* #35557 [quant][graphmode][fix] docs for InsertObservers
* **#35509 [quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`**

Summary:

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20742138](https://our.internmc.facebook.com/intern/diff/D20742138)

### CODE REVIEW

File: test/test_jit.py [hunk 1]
```diff
@@ -1612,6 +1612,36 @@ def forward(self, x, y):
...
+            m = torch.jit.script(M()).eval()
+            m = prepare_script(m, {'': default_qconfig}, True)
+            # for input tensor
+            assert len(attrs_with_prefix(m, '_observer')) == 1
+            data = torch.randn(1, 1, 10, 10, dtype=torch.float)
+            m(data)
+            m = convert_script(m, True)
+            FileCheck().check_not("aten::add") \
+                       .check_not("aten::add_") \
+                       .check_not("aten::relu") \
```
Comment (supriyar) [inline test/test_jit.py]: Should we also check for `quantized::add` , `quantized::relu`?

File: torch/csrc/jit/passes/quantization_patterns.h [hunk 2]
```diff
@@ -219,6 +253,7 @@ graph(%a_quant, %b_scalar, %alpha):
        add_scalar_out,
        quantized_add_scalar_out,
        add_scalar_filter},
+      {"quantized::cat", cat, quantized_cat},
```
Comment (supriyar) [inline torch/csrc/jit/passes/quantization_patterns.h]: is this change related to this PR?

**Review by supriyar — APPROVED** [review]
looks good. small comments.

File: torch/csrc/jit/passes/quantization_patterns.h (hunk 2 above)
Comment (jerryzh168) [reply torch/csrc/jit/passes/quantization_patterns.h]: this is just moving around quantized cat
